### PR TITLE
azure/s3: support downloading partial assets by range

### PR DESCRIPTION
**Pull request by jasonhancock**

### CODE REVIEW

File: test/test.go [hunk 1]
```diff
@@ -146,6 +146,15 @@ func All(t *testing.T, kind string, config stow.Config) {
 	is.Equal(readItemContents(is, item1), "item one")
 	is.NoErr(acceptableTime(t, is, items[0], item1))
 
+	if ir, ok := item1.(stow.ItemRanger); ok {
```
Comment (jdtobe) [inline test/test.go]: Does `item1` have the `ItemRanger` behavior to test here?

Comment (jasonhancock) [reply test/test.go]: It might, if it's an Azure location being tested.

Comment (jasonhancock) [reply test/test.go]: s/might/does/

**Comment by jasonhancock** [comment]
no with s3 support

**Pull request merged**